src/routes/login/LoginView.vue

```diff
@@ -1,393 +1,393 @@
+  </div>
+</template>
+
+<script>
+import { useI18n } from 'vue-i18n';
+import { mapStores } from 'pinia';
+import { useAuthStore } from '@/stores/auth';
+import { useThemedLogo } from '@/composables/themedLogo';
+import { ROUTE_NAME_LOGIN, ROUTE_NAME_SIGNUP } from '@/router';
+import BaseSnackbar from '@/components/BaseSnackbar.vue';
+import MarkdownContent from '@/components/MarkdownContent.vue';
+import BaseVideo from '@/components/BaseVideo.vue';
+import TheLanguageSelector from '@/components/TheLanguageSelector.vue';
+
+
+export default {
+  name: 'LoginView',
+  components: {
+    BaseSnackbar,
+    BaseVideo,
+    MarkdownContent,
+    TheLanguageSelector
+  },
+  setup() {
+    // Vuetify has its own translation layer that isn't very good
+    // It seems to conflict with the namespace of vue-i18n so we need to import it at the component
+    // level and utilize it this way with a composable. There may be another more global way but
+    // I haven't found anything just yet
+    const { t } = useI18n();
+    const { logoSrc } = useThemedLogo();
+    return {
+      t,
+      logoSrc,
+    };
+  },
+  data() {
+    return {
+      username: '',
+      pw: '',
+      showSnackBar: false,
+      snackBarMessage: '',
+      loading: false,
+      showPass: false,
+      isFormValid: false,
+      usernameRules: [this.isAlphaNumeric],
+      passwordRules: [this.has8OrMoreCharacters],
+    };
+  },
+  computed: {
+    ...mapStores(useAuthStore),
+    isLoggingIn() {
+      return this.$route.name === ROUTE_NAME_LOGIN;
+    },
+    formHeaderText() {
+      return this.isLoggingIn ? this.t('login.formHeaderLogin') : this.t('login.formHeaderSignup');
+    },
+    buttonText() {
+      if (this.isLoggingIn) {
+        return this.t('global.login');
+      }
+      return this.t('global.signup');
+    },
+    switchLabelText() {
+      if (this.isLoggingIn) {
+        return this.t('login.noAccount');
+      }
+      return this.t('login.haveAccount');
+    },
+  },
+  mounted() {
+    if (this.isLoggingIn) {
+      this.scrollAndFocusLogin();
+    }
+  },
+  methods: {
+    async submitLogin() {
+      this.loading = true;
+      const action = this.isLoggingIn ? this.authStore.requestLogin : this.authStore.requestSignup;
+      try {
+        await action({
+          username: this.username,
+          password: this.pw,
+        });
+        this.handleLogin();
+      } catch (err) {
+        this.handleError(err.message);
+      }
+    },
+    switchMode() {
+      this.pw = '';
+      if (this.isLoggingIn) {
+        this.$router.push({ name: ROUTE_NAME_SIGNUP });
+      } else {
+        this.$router.push({ name: ROUTE_NAME_LOGIN });
+      }
+      this.scrollAndFocusLogin();
+    },
+    handleLogin() {
+      this.username = '';
+      this.pw = '';
+      this.loading = false;
+      this.$router.push('/');
+    },
+    handleError(messageKey) {
+      this.showSnackBar = true;
+      this.snackBarMessage = messageKey;

```

Comment:
Let's move the translation here
```suggestion
      this.snackBarMessage = this.t(messageKey);
```